third_party/2and3/click/types.pyi

```diff
@@ -26,9 +28,17 @@ class FloatParamType(ParamType):
     def convert(self, value: str, param: Optional[Parameter], ctx: Optional[Context]) -> float: ...
 
 class FloatRange(FloatParamType):
+    min: Optional[float]
+    max: Optional[float]
+    clamp: bool
     def __init__(self, min: Optional[float] = ..., max: Optional[float] = ..., clamp: bool = ...) -> None: ...
 
 class File(ParamType):
+    mode: str
+    encoding: Optional[str]
+    errors: str
+    lazy: Optional[str]
+    atomic: bool
```

Comment:
Side note (not part of your PR): I just noticed that `errors` and `lazy` are marked as `Optional[]` in the `__init__()` argument, which seems to be wrong.

Reply:
I did a more thorough check now and it seems my change is wrong for `errors` while it defaults to the string `strict` the function calls it is used it actually allows for `None` to be passed in explicitly.

e.g., 
- [`_make_text_stream`](https://github.com/pallets/click/blob/7.1.2/src/click/_compat.py#L27)
- [`_get_windows_console_stream`](https://github.com/pallets/click/blob/7.1.2/src/click/_winconsole.py#L351)

Looking at `lazy` again it seems that the `__init__`  hint is correct and I made a mistake when reading the code the first time. From the [click doc](https://click.palletsprojects.com/en/7.x/arguments/#file-opening-safety) and its usage in the [code](https://github.com/pallets/click/blob/58a25939f0c4594dd5c487eec8906a79539e2810/src/click/types.py#L589) the attribute should be `Optional[bool]`.

Sorry I should have caught this in my first code review these click functions.
I will be changing the PR to

```python
errors: Optional[str]
lazy: Optional[bool]
```